tests/test_bm25.py

```diff
@@ -66,3 +66,26 @@ def test_bm25():
         rank_by=["Sum", [["blabla", "BM25", "walrus tusk"], ["blabla", "BM25", "jumping fox"]]]
     )
     assert [item.id for item in results] == [2, 6]
+
+    # Upsert with row-based upsert format
+    ns.upsert(
+        [
+            tpuf.VectorRow(id=8, vector=[0.8, 0.8], attributes={ "blabla": "row based upsert format is cool" }),
+        ],
+        schema=schema,
```

Comment:
Even though this seems to work, we should also make sure all the overrides have the schema in the parameter list